src/FlatSharp.Compiler/SchemaModel/ValueStructSchemaModel.cs

```diff
@@ -151,6 +155,16 @@ protected override void OnWriteCode(CodeWriter writer, CompileContext context)
 
             if (context.Options.GenerateMethods)
             {
+                string typeNames = string.Join(", ", this.fields.Select(x => x.TypeName));
+                string names = string.Join(", ", this.fields.Select(x => x.Name));
+                string tupleType = this.fields.Count == 0 ? "System.ValueTuple" : this.fields.Count == 1 ? $"System.ValueTuple<{typeNames}>" : $"({typeNames})";
+                string tupleValue = this.fields.Count < 2 ? $"System.ValueTuple.Create({names})" : $"({names})";
+                writer.AppendLine($"public {tupleType} ToTuple() => {tupleValue};");
```

Comment:
Do you think this should remain public?

Reply:
It is not a necessity for it to remain public. This can be private if that is preferred.

Reply:
I don't think I have big feelings about it. I just felt like I should leave some sort of comment :)